### ISSUE TITLE
[Bug]: WeaviateVectorStore never delete old documents after pipeline.run

### ISSUE THREAD

**Issue by stg609** [run-llama/llama_index#15014]
### Bug Description     
**TL;DR**        

When I run pipeline.run for a couple of times with `DocstoreStrategy.UPSERTS_AND_DELETE`, the object in weaviate is always increasing.     

---
I'm using `IngestionPipeline` with `WeaviateVectorStore`. The `pipeline.run` will invoke the `_handle_upserts` method. and eventually invoke `docstore.add_documents`. In the `add_documents` method, it will call the `_prepare_kv_pairs` to prepare a tuple named `metadata_kv_pairs` which uses `node.node_id`  rather than `node.ref_doc_id` as the first element . 
The next time when I run pipeline.run, the `_handle_upserts` will call below code to get a set of `node_id`, 
```
            existing_doc_ids_before = set(
                self.docstore.get_all_document_hashes().values()
            )
```
Then it will use the `node_id` as the `ref_doc_id`, and pass it to `vector_store.delete(ref_doc_id)`. The `WeaviateVectorStore.delete` will create a filter use below code which can't find any matching data.
```
where_filter = wvc.query.Filter.by_property("ref_doc_id").equal(ref_doc_id)
```

### Version

0.10.58

### init 
    vector_index_name = 'LA' 
    weaviate_client = weaviate.connect_to_wcs(
        cluster_url=settings.get("WEAVIATE_CLOUD_REST_URL"),
        auth_credentials=weaviate.auth.AuthApiKey(st.secrets.WEAVIATE_API_KEY),
    )

    vector_store = WeaviateVectorStore(
        weaviate_client=weaviate_client, index_name=vector_index_name,
    )

    docstore = SimpleDocumentStore()
    pipelineForTransforms = IngestionPipeline(     
        transformations=[
            SimpleFileNodeParser(),
            SentenceSplitter(chunk_size=500, chunk_overlap=100),
            Settings.embed_model
        ]
    ) 
    pipelineForStorage = IngestionPipeline(     
        transformations=[], # dont't need transform
        docstore = docstore,
        vector_store = vector_store,
        docstore_strategy = DocstoreStrategy.UPSERTS_AND_DELETE,
    ) 

### load data for a couple of times, you will see the number of objects is increasing.
    reader = FlatReader()

    docs = reader.load_data(Path("./data/xxx.md"))
    
    dir = os.listdir("./storage")
    if len(dir) != 0:
       _pipelineForStorage.load("./storage")

    
    nodes = _pipelineForTransform.run(documents=docs)

    # save to storage
    nodes = _pipelineForStorage.run(nodes=nodes)   # <------------ it will succeed, but weaviate won't delete old data

**Comment by logan-markewich** [comment]
@stg609 if the `id` of the incoming documents is not consistent, then it can't compare hashes to existing documents.

Ensure your input documents have the same document id for the same document (a filename for example is a good choice)

**Comment by stg609** [comment]
I replaced the old document with new one, so the id is not the same. However, that's what I think the  `DocstoreStrategy.UPSERTS_AND_DELET` should do to remove the old documents if not existed anymore. 
And actually, in the `_handle_upserts` method, it will remove the old documents from docstore as the delete_ref_doc in docstore actually take the ref_doc_id as doc_id. But the vector store (i.e. weaviate here) won't.
@logan-markewich
